### ISSUE TITLE
kops edit commands should be able to read from a file

### ISSUE THREAD

**Issue by zapman449** [kubernetes/kops#932]
Everything important should be under source control.  This includes the yaml files exposed via `kops edit cluster` and `kops edit instancegroup` etc.

The `kops edit` command should have a `--from-file` or `-f` flag which allows the administrator to feed in the new file from some source.  Bonus, the flag should take a hyphen (`-`) as the file name to signal that the file data comes from standard input.

**Comment by chrislovecnm** [comment]
Is this how kubectl edit works?

**Comment by zapman449** [comment]
kubectl has `apply` or `replace`

**Comment by zapman449** [comment]
kubectl edit also has:

```
Options:
  -f, --filename=[]: Filename, directory, or URL to file to use to edit the resource
```

**Comment by rlmcpherson** [comment]
I'm planning to implement this as suggested above, adding a `--filename` flag to the edit command, mirroring kubectl functionality. Is this a change that would be accepted upstream?

**Comment by justinsb** [comment]
Is `kubectl edit -f <file>` different from `kubectl replace -f <file>`?

I think the intention of replacing from a file is valid.  I had no idea kubectl supported `edit` from a file (that's weird!), but I guess if it is there, then yes we should support it too :-)

Contribution would be welcomed - thank you!

**Comment by rlmcpherson** [comment]
After looking at the kubectl behavior more, adding `replace`, as @justinsb  mentioned, is actually the correct thing to implement for non-interactive changes to resources. `kubectl edit -f <file>` still opens an editor, loading the file specified. It might still be useful to add a `kops edit --filename` flag but in order to mirror kubectl behavior it would still be interactive.

**Comment by rlmcpherson** [comment]
#1066 adds `kops replace` but it doesn't add a filename flag to the edit command. Hopefully it will address the issue for those wanting to check kops configuration files into source and update them non-interactively.

**Comment by justinsb** [comment]
I think this was implemented by #1066 - if we do have a separate need for `kops edit cluster -f` do comment / reopen.

And thanks for `kops replace` :-)
